app/controllers/gallery_controller.rb

```diff
@@ -7,7 +7,7 @@ def images
     if params[:query].present?
       results = FlickrService.new.search(params[:query])
 
-      images = results['photos']['photo']
+      images = results ? results['photos']['photo'] : []
```

Comment:
Can we use symbols here? I think you can also use the `dig` method

```ruby
images = results.dig(:photos, :photo)
```
If you still need an empty array, you can use a `||` here or in the render

Reply:
Thank you for your feedback, I didn't know the dig method, it's great. Also, I applied both recommendations.